docs/agents/grpc.md

```diff
@@ -0,0 +1,27 @@
+# gRPC support in agents
+
+## Header
+
+### Value format
+The value format of the header is text, as other vendors use text and there's no advantage to using a binary encoding. See technical details about the gRPC header [here](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests).
+
+### Key Names
+The header key names are `elastic-apm-traceparent` (for backwards compatibility with older agents) and `traceparent`.
+
+## Instrumented calls
+Server and Client Unary request/response calls are instrumented. Support for other calls may be added later (i.e. client/server streaming, bidirectional streaming).
+
+## Transaction/Span context schemas
+
+### Transaction context
+
+* **Name**: \<method\>, ex: `/helloworld.Greeter/SayHello`
+* **Type**: `request`
+* **Trace_context**: \<trace-context\>
+* **Result**: [\<a-valid-result-value\>](https://github.com/grpc/grpc/blob/master/doc/statuscodes.md#status-codes-and-their-use-in-grpc), ex: `OK`
```

Comment:
in the Java API, we have something that is called `authority` that provides `host:port` of the incoming transaction as a String.
1) is that something that is available to other implementations ?
2) probably more interesting to have it on the client side, as it provides host/port of the remote server

Reply:
I just checked the Ruby library and we can get a `peer` value containing `host:port` from a server call:
```ruby
(#<ElasticAPM::GRPC::ServerInterceptor>)> call
=> #<GRPC::ActiveCall::SingleReqView:0x00007f9ac27645f8
 @wrapped=
  #<GRPC::ActiveCall:0x00007f9ac2764c88
   @call=#<GRPC::Core::Call:0x00007f9ac27653b8>,
   @call_finished=false,
   @call_finished_mu=#<Thread::Mutex:0x00007f9ac2764af8>,
   @client_call_executed=false,
   @client_call_executed_mu=#<Thread::Mutex:0x00007f9ac2764ad0>,
   @deadline=1970-01-01 00:59:59 +0100,
   @input_stream_done=true,
   @marshal=#<Proc:0x00007f9ac2764f08@/Users/emily/.rvm/gems/ruby-2.6.5/gems/grpc-1.26.0-universal-darwin/src/ruby/lib/grpc/generic/rpc_desc.rb:35>,
   @metadata_received=true,
   @metadata_sent=false,
   @metadata_to_send={},
   @op_notifier=nil,
   @output_stream_done=false,
   @peer="ipv6:[::1]:58868",
   @peer_cert=nil,
   @send_initial_md_mutex=#<Thread::Mutex:0x00007f9ac2764be8>,
   @unmarshal=#<Proc:0x00007f9ac2764d78@/Users/emily/.rvm/gems/ruby-2.6.5/gems/grpc-1.26.0-universal-darwin/src/ruby/lib/grpc/generic/rpc_desc.rb:47>>>

[4] pry(#<ElasticAPM::GRPC::ServerInterceptor>)> call.peer
=> "ipv6:[::1]:58868"
```
The same `peer` value is not available on the public API from arguments to a client request:
```ruby
(#<ElasticAPM::GRPC::ClientInterceptor>)> call
=> #<GRPC::ActiveCall::InterceptableView:0x00007fc3fca106b8
 @wrapped=
  #<GRPC::ActiveCall:0x00007fc3fca10960
   @call=#<GRPC::Core::Call:0x00007fc3fca10a50>,
   @call_finished=false,
   @call_finished_mu=#<Thread::Mutex:0x00007fc3fca10820>,
   @client_call_executed=false,
   @client_call_executed_mu=#<Thread::Mutex:0x00007fc3fca107d0>,
   @deadline=1970-01-01 00:59:59 +0100,
   @input_stream_done=false,
   @marshal=#<Proc:0x00007fc3fb5cbf18@/Users/emily/.rvm/gems/ruby-2.6.5/gems/grpc-1.26.0-universal-darwin/src/ruby/lib/grpc/generic/rpc_desc.rb:35>,
   @metadata_received=false,
   @metadata_sent=false,
   @metadata_to_send={},
   @op_notifier=nil,
   @output_stream_done=false,
   @peer="localhost:50051",
   @send_initial_md_mutex=#<Thread::Mutex:0x00007fc3fca10870>,
   @unmarshal=#<Proc:0x00007fc3fb5cbe28@/Users/emily/.rvm/gems/ruby-2.6.5/gems/grpc-1.26.0-universal-darwin/src/ruby/lib/grpc/generic/rpc_desc.rb:47>>>

[3] pry(#<ElasticAPM::GRPC::ClientInterceptor>)> call.peer
NoMethodError: undefined method `peer' for #<GRPC::ActiveCall::InterceptableView:0x00007fc3fca106b8>
from (pry):3:in `request_response'
```
but I can still get it via Ruby backhand methods:
```ruby
(#<ElasticAPM::GRPC::ClientInterceptor>)> call.instance_variable_get(:@wrapped).peer
=> "localhost:50051"
```

What do you think @SylvainJuge ? I can include this as a field in the schema but say it's optional if your library doesn't expose an API to retrieve the value.

Reply:
Yes, let's make this optional, and for now only on the span.

Do we have an existing field that could be reused to store either `host:port` as-is, or then try to parse it and store it in two distinct fields ? If we go to parsing, we could use something close to ECS : https://www.elastic.co/guide/en/ecs/current/ecs-server.html#_server_field_details

Given it has a different name in Ruby and Java, it's likely to have other names in other platforms, thus splitting and parsing might help on this side to reuse existing fields.

Reply:
Could we use `destination.resource`? ref: https://github.com/elastic/apm/issues/180

Reply:
Also, for spans we should probably reuse those fields that are available to store this:
- `context.destination.address` (will require parsing)
- `context.destination.port` (will require parsing)

and the fields covered in https://github.com/elastic/apm/issues/180
- `context.destination.service.resource` in the `host:port` format
- `context.destination.service.name`
- `context.destination.service.type`

Reply:
ok, that sounds good. I'll update the spec to note these fields as optional.👍🏼